lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -163,55 +173,76 @@ def configure(conf)
+
+        this     = self
+        metadata = @cache.getset(cache_key) {
+          if metadata
+            kubernetes_metadata = this.get_metadata(
+              metadata['kubernetes']['namespace_name'],
+              metadata['kubernetes']['pod_name'],
+              metadata['kubernetes']['container_name']
+            )
+            metadata['kubernetes'] = kubernetes_metadata if kubernetes_metadata
+            metadata
+          end
+        }
+        if pod_id && (pod_id != metadata['kubernetes']['pod_id'])
+          log.debug("pod_id #{match_data['pod_id']} from log not equal to pod_id #{metadata['kubernetes']['pod_id']} from kubernetes for #{cache_key}")
+        end
+        if @include_namespace_id
+          namespace_name = metadata['kubernetes']['namespace_name']
+          namespace_id = @namespace_cache.getset(namespace_name) {
+            namespace = @client.get_namespace(namespace_name)
+            namespace['metadata']['uid'] if namespace
+          }
+          metadata['kubernetes']['namespace_id'] = namespace_id if namespace_id
+        end
+      end
+      metadata
+    end
+
     # NOTE: fluentd requires that records/hashes have string keys, not symbol keys
     # http://docs.fluentd.org/articles/plugin-development#record-format
     def filter_stream(tag, es)
       new_es = MultiEventStream.new
 
-      match_data = tag.match(@tag_to_kubernetes_name_regexp_compiled)
-
-      if match_data
-        metadata = {
-            'docker' => {
-                'container_id' => match_data['docker_id']
-            },
-            'kubernetes' => {
-                'namespace_name' => match_data['namespace'],
-                'pod_name'       => match_data['pod_name'],
-                'container_name' => match_data['container_name']
-            }
-        }
-
-        if @kubernetes_url.present?
-          cache_key = "#{metadata['kubernetes']['namespace_name']}_#{metadata['kubernetes']['pod_name']}_#{metadata['kubernetes']['container_name']}"
-
-          this     = self
-          metadata = @cache.getset(cache_key) {
-            if metadata
-              kubernetes_metadata = this.get_metadata(
-                metadata['kubernetes']['namespace_name'],
-                metadata['kubernetes']['pod_name'],
-                metadata['kubernetes']['container_name']
-              )
-              metadata['kubernetes'] = kubernetes_metadata if kubernetes_metadata
-              metadata
-            end
-          }
-          if @include_namespace_id
-            namespace_name = metadata['kubernetes']['namespace_name']
-            namespace_id = @namespace_cache.getset(namespace_name) {
-              namespace = @client.get_namespace(namespace_name)
-              namespace['metadata']['uid'] if namespace
-            }
-            metadata['kubernetes']['namespace_id'] = namespace_id if namespace_id
-          end
-        end
+      metadata = tag.match(@tag_to_kubernetes_name_regexp_compiled) do |match_data|
```

Comment:
My suggestion is to have a config parameter to decide whether the events are from journal or from log.
Then you can make the choice in `configure`, rather than when processing individual events.